src/test/java/org/apache/sysds/test/functions/builtin/BuiltinToOneHotTest.java

```diff
@@ -0,0 +1,126 @@
+import org.apache.sysds.test.TestConfiguration;
+import org.apache.sysds.test.TestUtils;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+
+import static org.junit.Assert.fail;
+
+public class BuiltinToOneHotTest extends AutomatedTestBase {
+    private final static String TEST_NAME = "to_one_hot";
+    private final static String TEST_DIR = "functions/builtin/";
+    private static final String TEST_CLASS_DIR = TEST_DIR + BuiltinToOneHotTest.class.getSimpleName() + "/";
+
+    private final static double eps = 0;
+    private final static int rows = 10;
+    private final static int cols = 1;
+    private final static int numClasses = 10;
+
+    @Override
+    public void setUp() {
+        addTestConfiguration(TEST_NAME,new TestConfiguration(TEST_CLASS_DIR, TEST_NAME, new String[]{"B"}));
+    }
+
+    @Test
+    public void runSimpleTest() {
+        runToOneHotTest(false, false, LopProperties.ExecType.CP, false);
+    }
+
+    @Test
+    public void runFailingSimpleTest() {
+        runToOneHotTest(false, false, LopProperties.ExecType.CP, true);
+    }
+
+    private void runToOneHotTest(boolean scalar, boolean sparse,
+                                 LopProperties.ExecType instType, boolean shouldFail) {
+        Types.ExecMode platformOld = setExecMode(instType);
+
+        try
+        {
+            loadTestConfiguration(getTestConfiguration(TEST_NAME));
+
+            //generate actual dataset
+            double[][] doubles = getRandomMatrix(rows, cols, 1, numClasses, 1, 7);
+
+            // round them
+            double[][] A = new double[rows][cols];
+            for(int i = 0; i < rows; i++) {
+                for(int j = 0; j < cols; j++) {
+                    A[i][j] = Math.round(doubles[i][j]);
+                }
+            }
+
+            int max = -1;
+
+            for(int i = 0; i < rows; i++) {
+                if(A[i][0] > max) {
+                    max = (int) A[i][0];
+                }
+            }
+
+            // script fails if numClasses provided is smaller than maximum value in A
+            int numClassesPassed = shouldFail ? max - 1 : max;
+
+            String HOME = SCRIPT_DIR + TEST_DIR;
+            fullDMLScriptName = HOME + TEST_NAME + ".dml";
+            programArgs = new String[]{"-explain", "-args", input("A"), String.format("%d", numClassesPassed),
+                    output("B") };
+
+
+            writeInputMatrixWithMTD("A", A, false);
+
+            runTest(true, false, null, -1);
+
+            if(!shouldFail) {
+                HashMap<MatrixValue.CellIndex, Double> expected = new HashMap<MatrixValue.CellIndex, Double>();
+                for(int i = 0; i < A.length; i++) {
+                    for(int j = 0; j < A[i].length; j++) {
+                        // indices start with 1 here
+                        expected.put(new MatrixValue.CellIndex(i + 1, (int) A[i][j]), 1.0);
+                    }
+                }
```

Comment:
maybe make this into an helper function to construct the result?
(the double for loop +  expected construction)